### PR TITLE
add support for require ldap-group

### PR DESCRIPTION
**Pull request by swenske**
#### Pull Request (PR) description
add support for require ldap-group

### CODE REVIEW

**Comment by swenske** [comment]
Hello @wyardley  !

https://github.com/voxpupuli/puppet-puppetboard/pull/173#issuecomment-330071461

I am interested for help , thanks

**Comment by wyardley** [comment]
Hi --

#voxpupuli on Freenode IRC or #voxpupuli / #testing on the Puppet slack are good places to get help with tests. Normally, when reworking something like this, it's ideal to preserve attribution to the original author in the commit (ideally by cherry-picking their commit), but I don't know that it's required.

**Comment by swenske** [comment]
hello @wyardley ,
I finally achieve to add tests and Travis seems to be happy with it.
Is it ok for you?

Thanks,
Sebastian

File: spec/acceptance/class_spec.rb [hunk 1]
```diff
@@ -126,6 +126,8 @@ class { 'puppetboard::apache::conf':
         ldap_bind_dn => 'cn=user,dc=puppet,dc=example,dc=com',
         ldap_bind_password => 'password',
         ldap_url     => 'ldap://puppet.example.com',
+        ldap_require_group => true,
```
Comment (wyardley) [inline spec/acceptance/class_spec.rb]: Hi - Yes, I think this is pretty close. But we should test both cases, so one context for ldap auth, and test that the config file does not contain "Require ldap-group".

Then make another conext that sets those parameters explicitly, and test for the expected results.

File: manifests/apache/vhost.pp [hunk 2]
```diff
@@ -88,6 +96,8 @@
   Optional[String] $ldap_bind_password      = undef,
   Optional[String] $ldap_url                = undef,
   Optional[String] $ldap_bind_authoritative = undef,
+  Boolean $ldap_require_group               = $::puppetboard::params::ldap_require_group,
```
Comment (wyardley) [inline manifests/apache/vhost.pp]: I believe the leading colons are not needed here anymore, so 
`$puppetboard::params...`

File: manifests/apache/conf.pp [hunk 3]
```diff
@@ -69,7 +77,9 @@
   Optional[String] $ldap_bind_dn            = undef,
   Optional[String] $ldap_bind_password      = undef,
   Optional[String] $ldap_url                = undef,
-  Optional[String] $ldap_bind_authoritative = undef
+  Optional[String] $ldap_bind_authoritative = undef,
+  Boolean $ldap_require_group               = $::puppetboard::params::ldap_require_group,
```
Comment (wyardley) [inline manifests/apache/conf.pp]: `$puppetboard::` vs `$::puppetboard`.

**Comment by wyardley** [comment]
I'd also acknowledge @rendhalver in the commit message if you can't get the commit directly attributed to him in your branch.

**Comment by swenske** [comment]
New PR [#236](https://github.com/voxpupuli/puppet-puppetboard/pull/236)

**Pull request closed without merging**